Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Ride.hs

```diff
@@ -13,74 +13,138 @@
 -}
 
 module Domain.Action.UI.Ride
-  ( GetDriverLocRes,
+  ( GetDriverLocResp,
+    GetRideStatusResp (..),
     getDriverLoc,
+    getRideStatus,
   )
 where
 
+import Domain.Types.Booking.BookingLocation (BookingLocationAPIEntity, makeBookingLocationAPIEntity)
+import qualified Domain.Types.Booking.Type as DB
 import qualified Domain.Types.Person as SPerson
 import Domain.Types.Ride
 import qualified Domain.Types.Ride as SRide
 import Environment
-import EulerHS.Prelude hiding (id)
+import Kernel.External.Encryption
 import qualified Kernel.External.Maps as Maps
+import Kernel.Prelude
+import Kernel.Storage.Esqueleto hiding (isNothing)
 import qualified Kernel.Storage.Hedis as Redis
 import Kernel.Types.Id
 import Kernel.Utils.CalculateDistance (distanceBetweenInMeters)
 import Kernel.Utils.Common
 import qualified SharedLogic.CallBPP as CallBPP
 import Storage.CachedQueries.CacheConfig
 import qualified Storage.Queries.Booking as QRB
+import qualified Storage.Queries.Person as QP
+import qualified Storage.Queries.Person.PersonDefaultEmergencyNumber as QPEN
 import qualified Storage.Queries.Ride as QRide
 import Tools.Error
 import qualified Tools.Maps as MapSearch
 import Tools.Metrics
 import qualified Tools.Notifications as Notify
 
-type GetDriverLocRes = MapSearch.LatLong
+type GetDriverLocResp = MapSearch.LatLong
+
+data GetRideStatusResp = GetRideStatusResp
+  { fromLocation :: BookingLocationAPIEntity,
+    toLocation :: Maybe BookingLocationAPIEntity,
+    ride :: RideAPIEntity,
+    customer :: SPerson.PersonAPIEntity,
+    driverPosition :: Maybe MapSearch.LatLong
+  }
+  deriving (Generic, FromJSON, ToJSON, Show, ToSchema)
 
 getDriverLoc ::
   ( HasCacheConfig r,
     EncFlow m r,
     EsqDBFlow m r,
+    EsqDBReplicaFlow m r,
     Redis.HedisFlow m r,
     CoreMetrics m,
     HasField "rideCfg" r RideConfig
   ) =>
   Id SRide.Ride ->
   Id SPerson.Person ->
-  m GetDriverLocRes
+  m GetDriverLocResp
 getDriverLoc rideId personId = do
-  ride <- QRide.findById rideId >>= fromMaybeM (RideDoesNotExist rideId.getId)
+  ride <- runInReplica $ QRide.findById rideId >>= fromMaybeM (RideDoesNotExist rideId.getId)
   when
     (ride.status == COMPLETED || ride.status == CANCELLED)
     $ throwError $ RideInvalidStatus "Cannot track this ride"
   res <- CallBPP.callGetDriverLocation ride
-  booking <- QRB.findById ride.bookingId >>= fromMaybeM (BookingDoesNotExist ride.bookingId.getId)
+  booking <- runInReplica $ QRB.findById ride.bookingId >>= fromMaybeM (BookingDoesNotExist ride.bookingId.getId)
+  unless (booking.riderId == personId) $ do
```

Comment:
We can remove this check too. We don't need to validate authToken. Just put api rate limit.

Reply:
As share ride will be open. Anyone should be able to access it.